src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -878,6 +878,7 @@ public synchronized void reset() {
                       + key.getIdentifier(),
                   e);
             } finally {
+              closedFuture.set(null);
```

Comment:
Given that closedFuture is assigned null, is there any chance that we can get here and NPE on closedFuture itself?

Reply:
Theoretically yes, if these two criteria are met.
1. At least two concurrent write requests #r1 and #r2 with the same digest but different write_id end up at the same worker simultaneously.
2.  The request #r1 creates an instance of Write but has not yet called the getOutput method (which is the next instruction after instance creation). Meanwhile, the request #r2 completes writing the digest. Consequently, it invokes the reset method through the listener on writeFuture before the request #r1 can call the getOutput method on the Write instance.

Not sure, this will ever happen. In any case added null check. Thanks for the feedback.